typedmodels/models.py

```diff
@@ -160,20 +162,18 @@ def do_related_class(self, other, cls):
             opts = cls._meta
 
             model_name = opts.model_name
-            typ = "%s.%s" % (opts.app_label, model_name)
+            typ = ContentType.objects.get(app_label=opts.app_label, model=model_name)
```

Comment:
```suggestion
            typ = ContentType.objects.get_for_model(cls)
```

Reply:
Huh, totally missed that. 

I've applied all of the suggestions, with one difference: as I understand `for_concrete_model=False` is also needed because of proxy the models.